File: webappdevelopment_ui/src/services/NotificationService.js
import http from "../http-common"; 

class NotificationService {
  getAllNotifications(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/notification/notifications`, searchDTO);
  }

  get(notificationId) {
    return this.getRequest(`/notification/${notificationId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/notification?field=${matchData}`, null);
  }

  addNotification(data) {
    return http.post("/notification/addNotification", data);
  }

  update(data) {
  	return http.post("/notification/updateNotification", data);
  }
  
  uploadImage(data,notificationId) {
  	return http.postForm("/notification/uploadImage/"+notificationId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new NotificationService();

File: src/main/java/com/webappdevelopment/service/NotificationService.java
package com.webappdevelopment.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.webappdevelopment.domain.Notification;
import com.webappdevelopment.dto.NotificationDTO;
import com.webappdevelopment.dto.NotificationSearchDTO;
import com.webappdevelopment.dto.NotificationPageDTO;
import com.webappdevelopment.dto.NotificationConvertCriteriaDTO;
import com.webappdevelopment.service.GenericService;
import com.webappdevelopment.dto.common.RequestDTO;
import com.webappdevelopment.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface NotificationService extends GenericService<Notification, Integer> {

	List<Notification> findAll();

	ResultDTO addNotification(NotificationDTO notificationDTO, RequestDTO requestDTO);

	ResultDTO updateNotification(NotificationDTO notificationDTO, RequestDTO requestDTO);

    Page<Notification> getAllNotifications(Pageable pageable);

    Page<Notification> getAllNotifications(Specification<Notification> spec, Pageable pageable);

	ResponseEntity<NotificationPageDTO> getNotifications(NotificationSearchDTO notificationSearchDTO);
	
	List<NotificationDTO> convertNotificationsToNotificationDTOs(List<Notification> notifications, NotificationConvertCriteriaDTO convertCriteria);

	NotificationDTO getNotificationDTOById(Integer notificationId);







}





